### PR TITLE
Only expand non-offstage widget

### PR DESCRIPTION
**Pull request by WenjiuWang**
| Before  | After |
| ------------- | ------------- |
| before_collapsed| after_collapsed  |
| before_expand  | after_expand  |

### CODE REVIEW

File: lib/ui/alert_detail.dart [hunk 1]
```diff
@@ -339,7 +339,9 @@ class _AlertDetailPageState extends State<AlertDetailPage> {
                     ].expand(
                           (widget) => [
                         widget,
-                        const SizedBox(
+                        (widget.runtimeType == Offstage && (widget as Offstage).offstage)
+                            ? const SizedBox.shrink()
```
Comment (kejuntong) [inline lib/ui/alert_detail.dart]: what if we do something like:

                          (widget) => (widget.runtimeType == Offstage && (widget as Offstage).offstage) ?
                            [widget] : [widget, const SizedBox(height: 24,)]

File: lib/ui/alert_detail.dart [hunk 2]
```diff
@@ -339,7 +339,9 @@ class _AlertDetailPageState extends State<AlertDetailPage> {
                     ].expand(
                           (widget) => [
                         widget,
-                        const SizedBox(
+                        (widget.runtimeType == Offstage && (widget as Offstage).offstage)
```
Comment (kejuntong) [inline lib/ui/alert_detail.dart]: Just out of curiosity, is the condition of `widget.runtimeType == Offstage` good enough to check whether it's offstage? Not too sure about the difference between the first condition and the second one :)

**Review by kejuntong — APPROVED** [review]

File: lib/ui/alert_detail.dart (hunk 2 above)
Comment (q58wu) [reply lib/ui/alert_detail.dart]: > Just out of curiosity, is the condition of `widget.runtimeType == Offstage` good enough to check whether it's offstage? Not too sure about the difference between the first condition and the second one :)

From reading the doc i think the first == Offstage check if for the type, second  .offstage is checking its state

File: lib/ui/alert_detail.dart (hunk 1 above)
Comment (q58wu) [reply lib/ui/alert_detail.dart]: > what if we do something like:
> 
> ```
>                       (widget) => (widget.runtimeType == Offstage && (widget as Offstage).offstage) ?
>                         [widget] : [widget, const SizedBox(height: 24,)]
> ```

i think that should work too! improves readability as well haha

**Review by q58wu — APPROVED** [review]

File: lib/ui/alert_detail.dart (hunk 2 above)
Comment (WenjiuWang) [reply lib/ui/alert_detail.dart]: Yeah the naming convention is bit weird, I'll never name a field the same as the class.

**Comment by WenjiuWang** [comment]
I applied formatting (ctrl alt l) so the code change appears to be more.

**Pull request merged**